crates/containerd-shim-wasmtime/src/executor.rs

```diff
@@ -0,0 +1,111 @@
+use nix::unistd::{dup, dup2};
+use std::{fs::OpenOptions, os::fd::RawFd};
+
+use anyhow::{anyhow, Result};
+use containerd_shim_wasm::sandbox::oci;
+use libc::{STDERR_FILENO, STDIN_FILENO, STDOUT_FILENO};
+use libcontainer::workload::{Executor, ExecutorError};
+use oci_spec::runtime::Spec;
+
+use wasmtime::{Engine, Linker, Module, Store};
+use wasmtime_wasi::WasiCtxBuilder;
+
+use crate::oci_wasmtime::{self, wasi_dir};
+
+const EXECUTOR_NAME: &str = "wasmtime";
+
+pub struct WasmtimeExecutor {
+    pub stdin: Option<RawFd>,
+    pub stdout: Option<RawFd>,
+    pub stderr: Option<RawFd>,
+    pub engine: Engine,
```

Comment:
I am not sure what benefits would it gain by using readonly at the cost of adding an extra dependency. Would you like to elaborate more about it?

Reply:
at current, they can be mutated by anybody inside or out of the module and crate. that seems like it could be problematic in the long term. another option, if you don't like the extra dependency, could be just make these module- or crate-private and provide getters as necessary.

Reply:
Ah I see. I am a bit hesitant because I believe this file / module is only used as a binary, not intended to use as a library, and hence mutation by anybody outside of the crate is less of a concern.